augur/curate/transform_strain_name.py

```diff
@@ -1,50 +1,77 @@
-#!/usr/bin/env python3
 """
-Verifies strain name pattern in the 'strain' field of the NDJSON record from
-stdin. Adds a 'strain' field to the record if it does not already exist.
-
-Outputs the modified records to stdout.
+Verifies strain name pattern in the 'strain' field of the NDJSON
+record. Adds a 'strain' field to the record if it does not already
+exist.
 """
+
 import argparse
-import json
 import re
-from sys import stderr, stdin, stdout
+from typing import Generator, List
+from augur.io.print import print_err
+from augur.utils import first_line
 
 
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter
-    )
-    parser.add_argument("--strain-regex", default="^.+$",
-        help="Regex pattern for strain names. " +
-             "Strain names that do not match the pattern will be dropped.")
-    parser.add_argument("--backup-fields", nargs="*",
-        help="List of backup fields to use as strain name if the value in 'strain' " +
-             "does not match the strain regex pattern. " +
-             "If multiple fields are provided, will use the first field that has a non-empty string.")
+def transform_name(
+    record: dict,
+    index: int,
+    strain_name_pattern: re.Pattern,
+    backup_fields: List[str],
```

Comment:
_nit_

`backup_fields` can potentially be `None` if no `--backup-fields` are provided right?

```suggestion
    backup_fields: Optional[List[str]],
```

OR

add a `default=[]` to `--backup-fields` argument.